src/include/firebird/impl/blr.h

```diff
@@ -346,6 +346,11 @@
 #define blr_trim_spaces		(unsigned char)0
 #define blr_trim_characters	(unsigned char)1
 
+/* multi-character trim for blr_trim */
+#define blr_trim_btrim		(unsigned char)3
+#define blr_trim_ltrim		(unsigned char)4
+#define blr_trim_rtrim		(unsigned char)5
+
```

Comment:
AFAIU, logically LTRIM works similarly to TRIM(LEADING), RTRIM - to TRIM(TRAILING), BTRIM - to TRIM(BOTH). If so, why do we need separate BLR verbs? If you need to distinguish between old and multi-character behaviour, then maybe just add something like:
`#define blr_trim_multi_characters (unsigned char)2`
instead ?

Reply:
The functions ltrim, rtrim, btrim have their own implementation different from the standard trim function, they remove all characters no matter in what order they are specified.
Example:
LTRIM('baobab is a three', 'oiba')
is a three

Reply:
Isn't it the way standard TRIM() was supposed to work?

Reply:
According to my old copy of old SQL 2008, standard TRIM allows single character values only in \<trim character\>:
```
11) If <trim function> is specified, then:
  a) Let S be the value of the <trim source>.
  b) If <trim character> is specified, then let SC be the value of <trim character>; otherwise, let SC be 
     <space>.
  c) If either S or SC is the null value, then the result of the <trim function> is the null value.
  d) If the length in characters of SC is not 1 (one), then an exception condition is raised: data exception —
     trim error.
```

Reply:
> AFAIU, logically LTRIM works similarly to TRIM(LEADING), RTRIM - to TRIM(TRAILING), BTRIM - to TRIM(BOTH). If so, why do we need separate BLR verbs? If you need to distinguish between old and multi-character behaviour, then maybe just add something like: `#define blr_trim_multi_characters (unsigned char)2` instead ?

I like it.

> The functions ltrim, rtrim, btrim have their own implementation different from the standard trim function, they remove all characters no matter in what order they are specified.

Modification of existing TrimNode is hard to name "own implementation" :)
And it have no relation with correct (better) usage of BLR verb's proposed above.

Reply:
> Isn't it the way standard TRIM() was supposed to work? At least in MS SQL it works this way.

Our `TRIM` allowed multi-character string incompatible with the new ones now defined in the standard.